cypress-io/cypress
# chore: update trigger workflow to use correct branch for internal contributions

- Closes N/A

### Additional details


I came across this issue while doing some related investigations into the data we collect from runs triggered by GitHub actions. I noticed in our [cloud project](https://cloud.cypress.io/projects/ypt4pf/branches) that we had a number of recordings matching a branch's internal ref (`pull/:num/head`) vs. the actual branch name. Traced it back to the workflow here, where we determine which branch to check out based on the contributor status. We're not looking up the repo name appropriately, which is forcing us to fall into the external contributor path all the time. This should clean up our Circle builds/Cypress recordings a bit and ensure that all builds are grouped under the expected branch.

Recent internal PR exhibiting problem: https://github.com/cypress-io/cypress/actions/runs/5605783165/job/15187030386#step:2:36

### Steps to test


~We'll validate the recorded run for this PR looks as expected.~ See [comment](https://github.com/cypress-io/cypress/pull/27357#issuecomment-1644390011)

### How has the user experience changed?


### PR Tasks


- [n/a] Have tests been added/updated?
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
